VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowConnection.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Meadow.CLI;
+using Meadow.Hcom;
+using System;
+using System.Threading;
+
+namespace Meadow
+{
+    static class MeadowConnection
+    {
+        static readonly int RETRY_COUNT = 10;
+        static readonly int RETRY_DELAY = 500;
+
+        internal static IMeadowConnection GetCurrentConnection()
```

Comment:
If we don't want to use SettingsManager.*.*  to pull in the last selected route, we might want to make the constructor signature something like:
`internal static IMeadowConnection GetCurrentConnection(string port, ILogger logger)`

Then construct the SerialConnection as:
`meadowConnection = new SerialConnection(port, logger);`

So that the SerialConnection can also report potential issues via the logger.